packages/ai-chat-ui/src/browser/chat-input-widget.tsx

```diff
@@ -268,30 +300,143 @@ const ChatInput: React.FunctionComponent<ChatInputProperties> = (props: ChatInpu
         }
     };
 
+    const leftOptions = React.useMemo<Option[]>(() => (
+        props.showContext ? [{
+            title: 'Attach elements to context',
+            handler: () => { /* TODO */ },
+            className: 'codicon-add'
+        }] : []
+    ), [props.showContext]);
+
+    const rightOptions = React.useMemo<Option[]>(() => (
```

Comment:
At the moment the memos are not worth it. The only thing we save are the construction of the arrays, however the `ChatInputOptions` will be rerendered anyway.

Only if we would `memo` the whole `ChatInputOptions`, then the `useMemo`s are needed. Not sure whether all these complications are worth it, I would imagine that the `ChatInputOptions` are cheap to render.